test/dotnet.Tests/CommandTests/ToolInstallGlobalOrToolPathCommandTests.cs

```diff
@@ -101,6 +101,28 @@ public void WhenPassingIgnoreFailedSourcesItShouldNotThrow()
             toolInstallGlobalOrToolPathCommand.Execute().Should().Be(0);
             _fileSystem.File.Delete(Path.Combine(_temporaryDirectory, "nuget.config"));
         }
+
+
+        [Fact]
+        public void WhenDuplicateSourceIsPassedIgnore()
+        {
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("NuGetConfigRandomPackageSources", allowCopyIfPresent: true)
+                .WithSource();
+
+            var packageSourceLocation = new PackageSourceLocation(
+                nugetConfig: new FilePath(Path.Combine(testAsset.Path, "NuGet.config")),
+                rootConfigDirectory: new DirectoryPath(testAsset.Path),
+                additionalSourceFeeds: ["https://api.nuget.org/v3/invalid.json"]);
+
+            var nuGetPackageDownloader = new NuGetPackageDownloader(new DirectoryPath(testAsset.Path));
+
+            var sources = nuGetPackageDownloader.LoadNuGetSources(new ToolPackage.PackageId(PackageId), packageSourceLocation);
+
+            sources.Where(s => s.SourceUri == new Uri("https://api.nuget.org/v3/invalid.json"))
```

Comment:
Is there a way to check that the source chosen is the one that came from NuGet.config?  Possibly by checking that the name is `invalid_source`.  This would ensure that the right one (ie the one which may have PAT, package source mapping, etc. associated with it) is chosen.

Reply:
Good idea, I added it, thanks!